data/blog/2022-04-08.robur.md

```diff
@@ -0,0 +1,47 @@
+
+Robur’s authoritative DNS server delegates responsibility for a specific domain and provides mapping information for it, ensuring that the user gets to the correct IP address. At the other end of the process, Robur’s DNS resolver finds the exact server to handle the user’s request. To keep the codebase minimal for security and simplicity, they included only the elements absolutely necessary. The let's encrypt unikernel waits for certificate requests in the zones (encoded as TLSA records), and uses let's encrypt DNS challenge to have these signed to certificates, again stored in the zones. Of course, certificates that will soon become invalid will be updated by dns-letsencrypt-secondary. This unikernel does not need any persistent storage, it leaves this to the dns-primary-git.
+
+We’ve been developing these DNS projects since 2017, which serve a multitude of functions in the Robur ecosystem. Our domains (like nqsb.io and robur.coop) use our DNS server as an authoritative server, and we also use a caching resolver for our biannual [MirageOS retreats](http://retreat.mirage.io) in Marrakech. Additionally, any MirageOS unikernel can use our client to resolve domain names - using the dns-client devices, or [happy-eyeballs](https://github.com/roburio/happy-eyeballs), so it’s also beneficial to the OCaml community at large. Robur uses expressive OCaml types (GADT), so we can ensure a given query has a specific shape, like an address record query results in a set of IPv4 addresses.
+
+Robur's DNS implementation can process extensions like dynamic updates, notifications, zone transfers, and request authentications. It can be installed through opam, as all OCaml tools and libraries, and you can find our DNS code’s [library](https://github.com/mirage/ocaml-dns) and unikernels [dns-primary-git](https://github.com/roburio/dns-primary-git), [dns-secondary](https://github.com/roburio/dns-secondary), [dns-letsencrypt](https://github.com/roburio/dns-letsencrypt-secondary), and [dns-resolver](https://git.robur.io/robur/dns-resolver) on GitHub.
```

Comment:
```suggestion
Robur's DNS implementation can process extensions like dynamic updates, notifications, zone transfers, and request authentications. It can be installed through opam, as all OCaml tools and libraries. You can find our DNS code’s [library](https://github.com/mirage/ocaml-dns) and unikernels ([`dns-primary-git`](https://github.com/roburio/dns-primary-git), [`dns-secondary`](https://github.com/roburio/dns-secondary), [`dns-letsencrypt`](https://github.com/roburio/dns-letsencrypt-secondary), and [`dns-resolver`](https://git.robur.io/robur/dns-resolver)) on GitHub.
```
This suggestion puts all the different unikernels in ( ) so they're not confused with being additions to the list starting with DNS code's library and uniknernels, that those are the unikernels. (If I understand correctly!)